src/Yarhl.UnitTests/IO/DataStreamTests.cs

```diff
@@ -1387,6 +1387,111 @@ public void WriteToAfterDispose()
             Assert.Throws<ObjectDisposedException>(() => stream1.WriteTo("/ex"));
         }
 
+        [Test]
+        public void WriteSegmentToVariableLength()
```

Comment:
Can you add another test to validate the overload `WriteSegmentTo(offset, stream)`, where offset is different to 0? This way we guarantee we are calculating the correct length always.